cli/src/main/kotlin/commands/UploadResultToSw360Command.kt

```diff
@@ -181,4 +217,43 @@ class UploadResultToSw360Command : CliktCommand(
 
     private fun createReleaseName(pkgId: Identifier) =
         listOf(pkgId.namespace, pkgId.name).filter { it.isNotEmpty() }.joinToString("/")
+
+    private fun uploadAttachmentToRelease(
+        sw360ReleaseClient: SW360ReleaseClientAdapter,
+        release: SW360Release,
+        path: Path,
+        type: SW360AttachmentType
+    ): SW360Release? {
+        val build = AttachmentUploadRequest.builder(release)
+            .addAttachment(path, type)
+            .build()
+
+        val uploadAttachmentResponse = sw360ReleaseClient.uploadAttachments(build)
+
+        return if (uploadAttachmentResponse.isSuccess) {
+            uploadAttachmentResponse.target.also {
+                log.debug("Uploaded attachment ${path.fileName} with type $type to SW360.")
+            }
+        } else {
+            release.also {
+                log.error("Could not upload attachment: " + uploadAttachmentResponse.failedUploads())
+            }
+        }
+    }
+
+    private fun saveSourcesTemporarily(pkg: Package, tempDirectory: File): File? {
+        val tempDownloadDirectory = tempDirectory.resolve("sources")
+        val download = Downloader(DownloaderConfiguration()).download(pkg, tempDownloadDirectory)
+
+        if (download.matches(pkg)) {
+            val zipFile = File(tempDirectory, "${pkg.id.toPath("-")}.zip")
+            tempDownloadDirectory.packZip(
+                zipFile,
+                directoryFilter = { it.name !in VCS_DIRECTORIES }
+            )
+
+            return zipFile
+        }
+        return null
```

Comment:
In case of an error, there should be a log statement that the sources for the package affected could not be uploaded.

Reply:
You mean not downloaded?

Reply:
Well, as a consequence of the failing download, the upload for this package - which is the purpose of this command - will fail, too.